### ISSUE TITLE
Notification badges in left panel contradict each other

### ISSUE THREAD

**Issue by kittykat** [element-hq/element-web#20372]
### Steps to reproduce

1. "Show all rooms" in Home is enabled.
2. Not reliably reproducible, just happens sometimes:

![Screenshot from 2021-12-16 19-03-43](https://user-images.githubusercontent.com/51663/147658501-8c6321ee-5f2b-4d35-94b2-59d99947a459.png)

#### What did you expect?
Home shows the culmination of notifications from all other 

#### What happened instead?
Sometimes I can see unread messages in a space, but not in home. Sometimes I can see a notification badge with a number in home, but not in any of the spaces when the triggering room is in a space. Sometimes clicking the notification badge over home doesn't take me to new messages.

Marking as S-Major because I can't always find the rooms which have new messages.

### Browser information

Chromium 96.0.4664.45 (Official Build) Arch Linux (64-bit)

### URL for webapp

develop.element.io

### Application version

Has been happening for a couple of weeks

### Will you send logs?

No

**Comment by t3chguy** [comment]
On your first screenshot, home includes orphans so the count is expected to be >= total

**Comment by kittykat** [comment]
> On your first screenshot, home includes orphans so the count is expected to be >= total

I just found the culprit: it was one of the rooms in the Element space

**Comment by kittykat** [comment]
I'm not sure if it's related, but I also have meta spaces telling me that I have unread messages but there aren't any in the room list:

![Screenshot from 2022-01-10 10-25-06](https://user-images.githubusercontent.com/51663/148751239-bf53eaef-8686-4d06-ac18-f1912db58a12.png)

**Comment by t3chguy** [comment]
The room list seems to think there are 84 too though

**Comment by kittykat** [comment]
Only the total for all favourites, not for any individual room in favourites

**Comment by t3chguy** [comment]
Yes, but that shows that the app is a lot more confused than just spaces being wrong
